wpimath/src/main/java/edu/wpi/first/math/autodiff/VariableBlock.java

```diff
@@ -0,0 +1,831 @@
+   * @param colSlice The column slice.
+   * @return A slice of the variable matrix.
+   */
+  public VariableBlock get(Slice.None rowSlice, Slice.None colSlice) {
+    return get(new Slice(rowSlice), new Slice(colSlice));
+  }
+
+  /**
+   * Returns a slice of the variable matrix.
+   *
+   * @param rowSlice The row slice.
+   * @param colSlice The column slice.
+   * @return A slice of the variable matrix.
+   */
+  public VariableBlock get(Slice.None rowSlice, Slice colSlice) {
+    return get(new Slice(rowSlice), colSlice);
+  }
+
+  /**
+   * Returns a slice of the variable matrix.
+   *
+   * @param rowSlice The row slice.
+   * @param colSlice The column slice.
+   * @return A slice of the variable matrix.
+   */
+  public VariableBlock get(Slice rowSlice, Slice.None colSlice) {
+    return get(rowSlice, new Slice(colSlice));
+  }
+
+  /**
+   * Returns a slice of the variable matrix.
+   *
+   * @param rowSlice The row slice.
+   * @param colSlice The column slice.
+   * @return A slice of the variable matrix.
+   */
+  public VariableBlock get(Slice rowSlice, Slice colSlice) {
+    int rowSliceLength = rowSlice.adjust(m_rowSliceLength);
+    int colSliceLength = colSlice.adjust(m_colSliceLength);
+    return new VariableBlock(
+        m_mat,
+        new Slice(
+            m_rowSlice.start + rowSlice.start * m_rowSlice.step,
+            m_rowSlice.start + rowSlice.stop * m_rowSlice.step,
+            rowSlice.step * m_rowSlice.step),
+        rowSliceLength,
+        new Slice(
+            m_colSlice.start + colSlice.start * m_colSlice.step,
+            m_colSlice.start + colSlice.stop * m_colSlice.step,
+            colSlice.step * m_colSlice.step),
+        colSliceLength);
+  }
+
+  /**
+   * Returns a block of the variable matrix.
+   *
+   * @param rowOffset The row offset of the block selection.
+   * @param colOffset The column offset of the block selection.
+   * @param blockRows The number of rows in the block selection.
+   * @param blockCols The number of columns in the block selection.
+   * @return A block of the variable matrix.
+   */
+  public VariableBlock block(int rowOffset, int colOffset, int blockRows, int blockCols) {
+    assert rowOffset >= 0 && rowOffset <= rows();
+    assert colOffset >= 0 && colOffset <= cols();
+    assert blockRows >= 0 && blockRows <= rows() - rowOffset;
+    assert blockCols >= 0 && blockCols <= cols() - colOffset;
+    return get(
+        new Slice(rowOffset, rowOffset + blockRows, 1),
+        new Slice(colOffset, colOffset + blockCols, 1));
+  }
+
+  /**
+   * Returns a segment of the variable vector.
+   *
+   * @param offset The offset of the segment.
+   * @param length The length of the segment.
+   * @return A segment of the variable vector.
+   */
+  public VariableBlock segment(int offset, int length) {
+    assert offset >= 0 && offset < rows() * cols();
+    assert length >= 0 && length <= rows() * cols() - offset;
+    return block(offset, 0, length, 1);
+  }
```

Comment:
If this is meant to be a segment of a vector, shouldn't this check `cols() == 1`?